lib/elastic_whenever/option.rb

```diff
@@ -133,10 +133,11 @@ def initialize(args)
                      elsif access_key && secret_key
                        Aws::Credentials.new(access_key, secret_key)
                      end
+      @credentials = nil unless @credentials&.set?
     end
 
     def aws_config
-      @aws_config ||= { credentials: credentials, region: region }.delete_if { |_k, v| v.nil? }
+      @aws_config ||= { credentials: credentials, region: region, profile: profile }.delete_if { |_k, v| v.nil? }
```

Comment:
Can you explain why this change is necessary? It looks like you need to specify the profile name, but my understanding is that passing `profile_name` in `Aws::SharedCredentials` should be an alternative to this.
https://docs.aws.amazon.com/sdk-for-ruby/v3/api/#Configuration_Options
https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/SharedCredentials.html

Reply:
with AWS SSO, I was not able to get the shared credentials to work as credentials was always `nil`. It's not necessary to set credentials at all in the case you are logged into SSO. The SDK should handle this as long as you have a profile set.

Reply:
Hmm, I did a little research and couldn't figure out why a profile need to be configured differently for SSO than others.
Is it not necessary to set `Aws::SSOCredentials` for SSO?
https://docs.aws.amazon.com/sdk-for-ruby/v3/developer-guide/credentials.html
https://docs.aws.amazon.com/sdkref/latest/guide/feature-sso-credentials.html
https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/sso.html
https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/SSOCredentials.html

Reply:
well, same with instance credentials, there's no need for explicit credentials when using SSO afaik

Reply:
at least this change works for me. we also successfully run elastic_whenever on AWS codebuild, where also no explicit credentials are present.

Reply:
if you prefer to pass explicit credentials, go for it, but IMO it's a no-op

Reply:
I could actually go around that change if I'd set `AWS_PROFILE`, I think! so should I revert the line?

Reply:
Ah, I understand. SSO uses `profile` regardless of shared credentials. I misunderstood that profile is a concept only for shared credentials.
Then this change makes sense. It's okay as is.